# Topic: Climate-Resilient Farming Practice

In [1]:
# Import libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [4]:
# Import Argricultural Survey data
raw_data = pd.read_csv('C:/Users/DONKAMS/Downloads/Hamoye-Premier-Project/Dataset/data.csv', low_memory=False)
pd.set_option('display.max_columns', None)
raw_data.head

<bound method NDFrame.head of       Unnamed: 0     hhcode  gender1  gender2  gender3  gender4  gender5  \
0              0   10010601      1.0      2.0      1.0      1.0      2.0   
1              1   10010602      1.0      2.0      1.0      1.0      1.0   
2              2   10010603      2.0      1.0      2.0      2.0      2.0   
3              3   10010604      2.0      2.0      2.0      1.0      1.0   
4              4   10010605      1.0      2.0      1.0      1.0      1.0   
...          ...        ...      ...      ...      ...      ...      ...   
9592        9592  100090113      1.0      2.0      2.0      1.0      1.0   
9593        9593  100090114      1.0      2.0      2.0      2.0      1.0   
9594        9594  100090115      1.0      2.0      1.0      2.0      2.0   
9595        9595  100090116      1.0      2.0      2.0      1.0      2.0   
9596        9596  100090117      1.0      1.0      1.0      2.0      2.0   

      gender6  gender7  gender8  age1  age2  age3  age4  

In [5]:
# Store names of columns externally to select needed columns
with open('columns.txt', 'w') as file:
    for item in raw_data.columns:
        file.write('%s\n' % item)

In [6]:
# Open the txt file in read mode
with open('columns_needed.txt', 'r') as file:
    # Read lines from the file and create a list
    columns_needed = [column.strip() for column in file.readlines()]
# Print the list of columns needed    
print(columns_needed)

['farmtype', 'fplots', 'fplotarea1', 'fplotarea2', 'fplotarea3', 'plotunits', 'fsystem1', 'fsystem2', 'fsystem3', 'fsystem4', 'tenure1', 'tenure2', 'tenure3', 'tenure4', 'yearsuse1', 'yearsuse2', 'yearsuse3', 'yearsuse4', 'rentplot1', 'rentplot2', 'rentplot3', 'rentplot4', 'farmbuyv', 'farmsalev', 'season1s', 'season1e', 'season2s', 'season2e', 'season3s', 'season3e', 'seas1nam', 'seas2nam', 'seas3nam', 's1p1c1', 's1p1c1plant', 's1p1c1harv', 's1p1c1area', 's1p1c1qharv', 's1p1c1cons', 's1p1c1lives', 's1p1c1lost', 's1p1c1mkt', 's1p1c1sold', 's1p1c1cval', 's1p1c1seed', 's1p1c1sval', 's1p1c2', 's1p1c2plant', 's1p1c2harv', 's1p1c2area', 's1p1c2qharv', 's1p1c2cons', 's1p1c2lives', 's1p1c2lost', 's1p1c2mkt', 's1p1c2sold', 's1p1c2cval', 's1p1c2seed', 's1p1c2sval', 's1p1c3', 's1p1c3plant', 's1p1c3harv', 's1p1c3area', 's1p1c3qharv', 's1p1c3cons', 's1p1c3lives', 's1p1c3lost', 's1p1c3mkt', 's1p1c3sold', 's1p1c3cval', 's1p1c3seed', 's1p1c3sval', 's1p1c4', 's1p1c4plant', 's1p1c4harv', 's1p1c4area', 

In [7]:
# Filter raw data to contain columns needed
filtered_data = raw_data[columns_needed]
# Print filtered_data
print(filtered_data)

      farmtype  fplots  fplotarea1  fplotarea2  fplotarea3 plotunits  \
0          2.0     1.0         5.0        0.00         0.0        HA   
1          2.0     1.0         5.0        3.00         0.0        HA   
2          2.0     1.0         3.0        4.00         0.0        HA   
3          1.0     1.0         1.0        0.50         0.0        HA   
4          2.0     1.0         5.0        1.75         0.0        HA   
...        ...     ...         ...         ...         ...       ...   
9592       2.0     1.0         3.0        0.00         0.0        HA   
9593       2.0     1.0         3.0        0.00         0.0        HA   
9594       2.0     1.0         1.0        0.00         0.0        HA   
9595       1.0     1.0         1.0        0.00         0.0        HA   
9596       2.0     1.0         2.0        0.00         0.0        HA   

      fsystem1  fsystem2  fsystem3  fsystem4  tenure1  tenure2  tenure3  \
0          2.0       NaN       NaN       NaN      1.0      N

In [8]:
# Calculate sum of missing data in each column
filtered_data.isna().sum()

farmtype                 109
fplots                    29
fplotarea1                 0
fplotarea2                 0
fplotarea3                 0
                        ... 
adjtempshifts2_1        9554
adjtempshifts3_1        9586
adjrainfallshifts1_1    8938
adjrainfallshifts2_1    9535
adjrainfallshifts3_1    9580
Length: 823, dtype: int64

In [9]:
# Calculate the percentage of NaN values in each column
na_percentage = (filtered_data.isna().mean() * 100)

# Drop columns with more than 80% NaN values
columns_to_drop = na_percentage[na_percentage > 80].index
filtered_data = filtered_data.drop(columns=columns_to_drop)

# Store data of columns droped
with open('columns_droped.txt', 'w') as file:
    for item in columns_to_drop:
        file.write('%s\n' % item)
        
print(filtered_data.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9597 entries, 0 to 9596
Columns: 636 entries, farmtype to ad7625
dtypes: float64(519), object(117)
memory usage: 46.6+ MB
None


In [10]:
# Calculate threshold for no of na values in each row, 80%
threshold = 0.8 * len(filtered_data.columns)

# Number of rows to be dropped
dropped_indices = filtered_data.index[filtered_data.isnull().sum(axis=1) > threshold].tolist()
print(len(dropped_indices))

# Drop rows with more than 80% NaN values
filtered_data = filtered_data.dropna(thresh=threshold)

273


In [11]:
filtered_data.describe()

farmtype       fplots   fplotarea1   fplotarea2   fplotarea3  \
count  1800.000000  1800.000000  1800.000000  1800.000000  1800.000000   
mean      1.725556     1.885556     3.775825     1.164083     0.533577   
std       0.739734     1.007041     6.508500     2.204666     1.942052   
min       1.000000     1.000000     0.000100     0.000000     0.000000   
25%       1.000000     1.000000     0.625000     0.000000     0.000000   
50%       2.000000     2.000000     2.083300     0.208000     0.000000   
75%       2.000000     2.000000     5.000000     1.500000     0.000000   
max       3.000000    10.000000   150.000000    40.000000    40.000000   

          fsystem1     fsystem2      tenure1      tenure2    yearsuse1  \
count  1800.000000  1800.000000  1800.000000  1800.000000  1800.000000   
mean      2.403889     1.136111     1.443889     0.795000    24.869167   
std       0.729499     1.304129     1.314462     1.432831    24.319732   
min       0.000000     0.000000     0.000000     0.000000     0.000000   
25%       2.000000     0.000000     1.000000     0.000000    10.000000   
50%       2.000000     0.000000     1.000000     0.000000    20.000000   
75%       3.000000     2.000000     1.000000     1.000000    32.000000   
max       5.000000     6.000000     7.000000    20.000000   202.000000   

         yearsuse2     rentplot1      rentplot2      farmbuyv     farmsalev  \
count  1800.000000   1800.000000    1800.000000  1.800000e+03  1.800000e+03   
mean     10.847778    185.172067     215.452250  3.253343e+05  6.547597e+05   
std      21.132319   1414.373995    5184.715254  9.743351e+05  2.100542e+06   
min       0.000000      0.000000       0.000000  0.000000e+00  0.000000e+00   
25%       0.000000      0.000000       0.000000  0.000000e+00  5.045000e+04   
50%       0.000000      0.000000       0.000000  0.000000e+00  1.862500e+05   
75%      15.000000      0.000000       0.000000  2.750000e+05  5.760000e+05   
max     200.000000  20000.000000  210000.000000  2.000000e+07  4.680000e+07   

            s1p1c1   s1p1c1area   s1p1c1qharv    s1p1c1cons   s1p1c1lives  \
count  1655.000000  1655.000000  1.800000e+03   1800.000000  1.800000e+03   
mean     31.143202    63.184949  1.239530e+04    924.509444  6.559703e+03   
std      15.932655    24.126652  1.041338e+05   1868.210249  1.024222e+05   
min       3.000000     4.550000  0.000000e+00      0.000000  0.000000e+00   
25%      27.000000    50.000000  5.482500e+02      0.000000  0.000000e+00   
50%      27.000000    60.000000  1.777500e+03    450.000000  0.000000e+00   
75%      54.000000    80.000000  5.010000e+03   1076.250000  0.000000e+00   
max      56.000000   100.000000  4.200000e+06  31258.000000  4.200000e+06   

          s1p1c1lost    s1p1c1mkt     s1p1c1sold    s1p1c1cval       s1p1c2  \
count    1800.000000  1800.000000    1800.000000  6.890000e+02  1052.000000   
mean      130.606944     0.764444    5267.469167  3.088056e+04    24.739544   
std      4427.437852     0.969268   28162.823868  9.730573e+04    16.204704   
min         0.000000     0.000000       0.000000  7.500000e+01     3.000000   
25%         0.000000     0.000000       0.000000  1.100000e+03    12.000000   
50%         0.000000     0.000000       0.000000  2.441000e+03    12.000000   
75%         0.000000     2.000000    1800.000000  9.350000e+03    44.000000   
max    187500.000000     3.000000  900000.000000  1.600000e+06    56.000000   

        s1p1c2area   s1p1c2qharv    s1p1c2cons   s1p1c2lives   s1p1c2lost  \
count  1052.000000  1.800000e+03   1800.000000  1.800000e+03  1800.000000   
mean     25.464030  3.921193e+03    198.123556  2.865899e+03    17.060278   
std      15.152534  5.823022e+04    581.664183  5.806850e+04   113.386726   
min       2.000000  0.000000e+00      0.000000  0.000000e+00     0.000000   
25%      15.000000  0.000000e+00      0.000000  0.000000e+00     0.000000   
50%      20.000000  5.000000e+01      0.000000  0.000000e+00     0.000000   
75%      33.330002 

In [13]:
filtered_data.shape

(1800, 636)

So means have been made to reduce this to 636 columns, great I will be working with that then. below are the procedures I am following to preprocess and clean the dataset 
1. Create an empty dataframe to acquire all newly processed column
2. create a list for columns having homogeuity
3. Through documentation reading and research a means is devised to aggregrate the columns into one or two
4. The newly created column which has same information as those columns is added to the new dataframe

In [14]:
# Create an empty dataframe
df = pd.DataFrame()
df.shape

(0, 0)

I will be using the columns needed list as my base columns to work with

In [18]:
#checkig out the first 22 columns
first_22_columns = filtered_data.iloc[:, :22]

In [19]:
first_22_column_names = filtered_data.columns[:22]

In [20]:
first_22_column_names

Index(['farmtype', 'fplots', 'fplotarea1', 'fplotarea2', 'fplotarea3',
       'plotunits', 'fsystem1', 'fsystem2', 'tenure1', 'tenure2', 'yearsuse1',
       'yearsuse2', 'rentplot1', 'rentplot2', 'farmbuyv', 'farmsalev',
       'season1s', 'season1e', 'season2s', 'season2e', 'season3s', 'season3e'],
      dtype='object')

In [21]:
first_22_columns

farmtype  fplots  fplotarea1  fplotarea2  fplotarea3 plotunits  \
1087       3.0     1.0      2.5000      0.0000      0.0000        HA   
1088       2.0     3.0      0.5042      0.4500      0.4875        HA   
1089       2.0     3.0      0.4500      0.5208      0.5542        HA   
1090       2.0     3.0      0.5542      0.5542      0.5542        HA   
1091       1.0     2.0      0.4167      0.2083      0.0000        HA   
...        ...     ...         ...         ...         ...       ...   
5474       2.0     1.0      5.0000      0.0000      0.0000        HA   
5475       2.0     1.0      8.0000      0.0000      0.0000        HA   
5476       2.0     2.0     11.0000      6.0000      0.0000        HA   
5477       2.0     1.0      8.0000      0.0000      0.0000        HA   
5478       2.0     2.0      8.0000      5.0000      0.0000        HA   

      fsystem1  fsystem2  tenure1  tenure2  yearsuse1  yearsuse2  rentplot1  \
1087       3.0       0.0      1.0      0.0       16.0        0.0        0.0   
1088       2.0       2.0      1.0      1.0       33.0       33.0        0.0   
1089       3.0       3.0      1.0      1.0       33.0       33.0        0.0   
1090       3.0       3.0      1.0      1.0       33.0       33.0        0.0   
1091       2.0       2.0      1.0      1.0       25.0       25.0        0.0   
...        ...       ...      ...      ...        ...        ...        ...   
5474       3.0       0.0      1.0      0.0        9.0        0.0        0.0   
5475       1.0       0.0      1.0      0.0       11.0        0.0        0.0   
5476       3.0       2.0      1.0      1.0       26.0       18.0        0.0   
5477       3.0       0.0      1.0      0.0        8.0        0.0        0.0   
5478       3.0       2.0      1.0      1.0       15.0       12.0     2000.0   

      rentplot2  farmbuyv  farmsalev   season1s   season1e   season2s  \
1087        0.0       0.0   150000.0  w1oct2001  w1may2002  w1may2002   
1088        0.0       0.0   175000.0  w1oct2001  w1may2002  w1may2002   
1089        0.0       0.0   210000.0  w1oct2001  w1may2002  w1may2002   
1090        0.0       0.0   240000.0  w1oct2001  w1may2002  w1may2002   
1091        0.0       0.0    30000.0  w1oct2001  w1may2002  w1may2002   
...         ...       ...        ...        ...        ...        ...   
5474        0.0       0.0   200000.0      1-Jun     30-Sep      1-Nov   
5475        0.0       0.0   400000.0      1-Jun     30-Sep      1-Nov   
5476        0.0       0.0   600000.0      1-Jun     30-Sep      1-Nov   
5477        0.0       0.0   300000.0      1-Jun     30-Sep      1-Nov   
5478        0.0       0.0   500000.0      1-Jun     30-Sep      1-Nov   

       season2e   season3s   season3e  
1087  w1oct2002  w1aug2002  w4dec2002  
1088  w1oct2002  w1aug2002  w4dec2002  
1089  w1oct2002  w1aug2002  w4dec2002  
1090  w1oct2002  w1aug2002  w4dec2002  
1091  w1oct2002  w1aug2002  w4dec2002  
...         ...        ...        ...  
5474     31-Mar      1-Dec      1-May  
5475     31-Mar      1-Dec      1-May  
5476     31-Mar      1-Dec      1-May  
5477     31-Mar      1-Dec      1-May  
5478     31-Mar      1-Dec      1-May  

[1800 rows x 22 columns]

1. for the farm type, this is an important key which is a labelled data that is of immense importance in the study
2. The classifcation of family plots is also a significant but not really needed feature in this study
3. The fplotareas represent the area of plot of land for the families, To represent the area of land for families more efficiently, we can combine the three columns into a single column. We create a new column called "TotalPlotArea" to store the sum of the individual plot areas for each family.
4. We combined related information into new columns, making it easier to understand and analyze the data.
 'LandSystem' now represents the land system information, combining fsystem1 and fsystem2.
 'TenureType' summarizes the tenure information from tenure1 and tenure2.
 'UsageYears' provides a concise description of years of usage.

In [23]:
df['farmtype'] = filtered_data['farmtype']
# Create a new column 'TotalPlotArea' that sums the individual plot areas
df['TotalPlotArea'] = filtered_data['fplotarea1'] + filtered_data['fplotarea2'] + filtered_data['fplotarea3']

# Combine fsystem1 and fsystem2 into a new column 'LandSystem'
df['LandSystem'] = filtered_data['fsystem1'].astype(str) + ' ' + filtered_data['fsystem2'].astype(str)

# Combine tenure1 and tenure2 into a new column 'TenureType'
df['TenureType'] = filtered_data['tenure1'].astype(str) + ' ' + filtered_data['tenure2'].astype(str)

# Combine yearsuse1 and yearsuse2 into a new column 'UsageYears'
df['UsageYears'] = filtered_data['yearsuse1'].astype(str) + ' ' + filtered_data['yearsuse2'].astype(str)

In [24]:
df

farmtype  TotalPlotArea LandSystem TenureType UsageYears
1087       3.0         2.5000    3.0 0.0    1.0 0.0   16.0 0.0
1088       2.0         1.4417    2.0 2.0    1.0 1.0  33.0 33.0
1089       2.0         1.5250    3.0 3.0    1.0 1.0  33.0 33.0
1090       2.0         1.6626    3.0 3.0    1.0 1.0  33.0 33.0
1091       1.0         0.6250    2.0 2.0    1.0 1.0  25.0 25.0
...        ...            ...        ...        ...        ...
5474       2.0         5.0000    3.0 0.0    1.0 0.0    9.0 0.0
5475       2.0         8.0000    1.0 0.0    1.0 0.0   11.0 0.0
5476       2.0        17.0000    3.0 2.0    1.0 1.0  26.0 18.0
5477       2.0         8.0000    3.0 0.0    1.0 0.0    8.0 0.0
5478       2.0        13.0000    3.0 2.0    1.0 1.0  15.0 12.0

[1800 rows x 5 columns]

In [28]:
#checking out columns from index 23 to 501
second_23_200_columns = filtered_data.iloc[:, 23:200]
second_23_200_columns_names = filtered_data.columns[23:200]

In [29]:
second_23_200_columns_names

Index(['seas2nam', 'seas3nam', 's1p1c1', 's1p1c1plant', 's1p1c1harv',
       's1p1c1area', 's1p1c1qharv', 's1p1c1cons', 's1p1c1lives', 's1p1c1lost',
       ...
       's2p1c4plant', 's2p1c4harv', 's2p1c4qharv', 's2p1c4cons', 's2p1c4lives',
       's2p1c4lost', 's2p1c4mkt', 's2p1c4sold', 's2p1c4seed', 's2p1c4sval'],
      dtype='object', length=177)

In [30]:
second_23_200_columns

seas2nam seas3nam  s1p1c1 s1p1c1plant s1p1c1harv  s1p1c1area  \
1087        2        3    54.0   15Nov2001  15May2002   33.330002   
1088        2        3    54.0   15Nov2001  15May2002   34.970001   
1089        2        3    54.0   15Nov2001  15May2002   29.510000   
1090        2        3    54.0   15Nov2001  15May2002   33.330002   
1091        2        3    54.0   15Nov2001  15May2002   66.669998   
...       ...      ...     ...         ...        ...         ...   
5474        1        3    27.0       1-Jun     30-Sep   90.000000   
5475        1        3    26.0       1-Jun     30-Sep   50.000000   
5476        1        3    27.0       1-Jun     30-Sep   80.000000   
5477        1        3    27.0       1-Jun     30-Sep   50.000000   
5478        1        3    27.0       1-Jun     30-Sep   80.000000   

      s1p1c1qharv  s1p1c1cons  s1p1c1lives  s1p1c1lost  s1p1c1mkt  s1p1c1sold  \
1087       4800.0     24000.0          0.0         0.0        2.0      2400.0   
1088       2480.0      1550.0          0.0         0.0        2.0       930.0   
1089       2700.0      2000.0          0.0         0.0        2.0       700.0   
1090       2250.0      1250.0          0.0         0.0        2.0      1000.0   
1091       2250.0      1250.0          0.0         0.0        2.0      1000.0   
...           ...         ...          ...         ...        ...         ...   
5474       2000.0       400.0          0.0         0.0        2.0      1600.0   
5475       2800.0       600.0          0.0         0.0        2.0      2200.0   
5476       3200.0      1200.0          0.0         0.0        2.0      2000.0   
5477       2000.0       600.0          0.0         0.0        2.0      1400.0   
5478       3000.0       600.0          0.0         0.0        2.0      2400.0   

      s1p1c1cval s1p1c1seed   s1p1c1sval  s1p1c2 s1p1c2plant s1p1c2harv  \
1087      1710.0        140            2     9.0   10Sep2001  08Apr2002   
1088       720.0         60          2.5     9.0   10Sep2001  01May2002   
1089       560.0         60  2.329999924    38.0   01Jan2002  30Apr2002   
1090       800.0         60          2.5     9.0   10Sep2001  01May2002   
1091      1000.0         60            2     9.0   17Sep2001  08May2002   
...          ...        ...          ...     ...         ...        ...   
5474    112000.0         10          425    12.0       1-Jun     30-Sep   
5475    110000.0         10          375    29.0       1-Jun     30-Sep   
5476    140000.0         20          425    12.0       1-Jun     30-Sep   
5477     98000.0         10          450    12.0       1-Jun     30-Sep   
5478    160000.0         16          475    12.0       1-Jun     30-Sep   

      s1p1c2area  s1p1c2qharv  s1p1c2cons  s1p1c2lives  s1p1c2lost  s1p1c2mkt  \
1087   33.330002      60000.0         0.0      60000.0         0.0        0.0   
1088   31.209999      35000.0         0.0      35000.0         0.0        0.0   
1089   34.150002      15000.0       500.0          0.0         0.0        2.0   
1090   33.330002      38000.0         0.0      38000.0         0.0        0.0   
1091   33.330002      14000.0         0.0      14000.0         0.0        0.0   
...          ...          ...         ...          ...         ...        ...   
5474   10.000000        600.0       200.0          0.0       100.0        2.0   
5475   50.000000       3200.0       200.0          0.0         0.0        2.0   
5476   20.000000       1000.0       300.0          0.0       200.0        2.0   
5477   50.000000       1600.0       200.0          0.0       100.0        2.0   
5478   20.000000       1000.0       300.0          0.0       100.0        2.0   

      s1p1c2sold  s1p1c2cval s1p1c2seed   s1p1c2sval  s1p1c3 s1p1c3plant  \
1087         0.0         NaN         50            4     4.0   04Nov2001   
1088         0.0         NaN         30            4    38.0   01Jan2002   
1089     14500.0      9400.0        750  5.329999924     9.0   10Sep2001   
1090         0.0         

In [37]:
#check for missing values in each columns
second_23_200_columns.isnull().any().sum()

# Check for missing values in each column
columns_with_missing_values = second_23_200_columns.columns[second_23_200_columns.isnull().any()]

# Drop columns with missing values
second_23_200_columns_dropped = second_23_200_columns.drop(columns=columns_with_missing_values, axis=1)

# second_23_200_columns_dropped now contains only columns without missing values

In [38]:
second_23_200_columns_dropped

seas2nam seas3nam  s1p1c1qharv  s1p1c1cons  s1p1c1lives  s1p1c1lost  \
1087        2        3       4800.0     24000.0          0.0         0.0   
1088        2        3       2480.0      1550.0          0.0         0.0   
1089        2        3       2700.0      2000.0          0.0         0.0   
1090        2        3       2250.0      1250.0          0.0         0.0   
1091        2        3       2250.0      1250.0          0.0         0.0   
...       ...      ...          ...         ...          ...         ...   
5474        1        3       2000.0       400.0          0.0         0.0   
5475        1        3       2800.0       600.0          0.0         0.0   
5476        1        3       3200.0      1200.0          0.0         0.0   
5477        1        3       2000.0       600.0          0.0         0.0   
5478        1        3       3000.0       600.0          0.0         0.0   

      s1p1c1mkt  s1p1c1sold s1p1c1seed   s1p1c1sval  s1p1c2qharv  s1p1c2cons  \
1087        2.0      2400.0        140            2      60000.0         0.0   
1088        2.0       930.0         60          2.5      35000.0         0.0   
1089        2.0       700.0         60  2.329999924      15000.0       500.0   
1090        2.0      1000.0         60          2.5      38000.0         0.0   
1091        2.0      1000.0         60            2      14000.0         0.0   
...         ...         ...        ...          ...          ...         ...   
5474        2.0      1600.0         10          425        600.0       200.0   
5475        2.0      2200.0         10          375       3200.0       200.0   
5476        2.0      2000.0         20          425       1000.0       300.0   
5477        2.0      1400.0         10          450       1600.0       200.0   
5478        2.0      2400.0         16          475       1000.0       300.0   

      s1p1c2lives  s1p1c2lost  s1p1c2mkt  s1p1c2sold s1p1c2seed   s1p1c2sval  \
1087      60000.0         0.0        0.0         0.0         50            4   
1088      35000.0         0.0        0.0         0.0         30            4   
1089          0.0         0.0        2.0     14500.0        750  5.329999924   
1090      38000.0         0.0        0.0         0.0         30  4.329999924   
1091      14000.0         0.0        0.0         0.0         15         4.75   
...           ...         ...        ...         ...        ...          ...   
5474          0.0       100.0        2.0       300.0          4          400   
5475          0.0         0.0        2.0      3000.0         10          500   
5476          0.0       200.0        2.0       500.0          6          400   
5477          0.0       100.0        2.0      1300.0         10          400   
5478          0.0       100.0        2.0       700.0          5          425   

      s1p1c3cons  s1p1c3lost  s1p1c3mkt  s1p1c3sold s1p1c3seed   s1p1c3sval  \
1087         0.0         0.0        2.0      2550.0         50            3   
1088         0.0         0.0        2.0     13500.0        700  2.700000048   
1089         0.0         0.0        0.0         0.0         30  4.329999924   
1090       500.0         0.0        2.0     17000.0        750  2.329999924   
1091         0.0         0.0        0.0         0.0          0            0   
...          ...         ...        ...         ...        ...          ...   
5474         0.0         0.0        0.0         0.0          0            0   
5475         0.0         0.0        0.0         0.0          0            0   
5476         0.0         0.0        0.0         0.0          0            0   
5477         0.0         0.0        0.0         0.0          0            0   
5478         0.0         0.0        0.0         0.0          0            0   

      s1p1c4qharv  s1p1c4cons  s1p1c4lives  s1p1c4lost  s1p1c4mkt  s1p1c4sold  \
1087          0.0         0.0          0.0         0.0        0.0         0.0   
1088          0.0         0.0          0.0         0.0     

In [39]:
second_23_200_columns_dropped.columns

Index(['seas2nam', 'seas3nam', 's1p1c1qharv', 's1p1c1cons', 's1p1c1lives',
       's1p1c1lost', 's1p1c1mkt', 's1p1c1sold', 's1p1c1seed', 's1p1c1sval',
       ...
       's2p1c3seed', 's2p1c3sval', 's2p1c4qharv', 's2p1c4cons', 's2p1c4lives',
       's2p1c4lost', 's2p1c4mkt', 's2p1c4sold', 's2p1c4seed', 's2p1c4sval'],
      dtype='object', length=114)

In [40]:
selected_columns = filtered_data[
    ['seas2nam', 'seas3nam',
     's1p1c1qharv', 's1p1c1cons', 's1p1c1lives', 's1p1c1lost', 's1p1c1mkt', 's1p1c1sold', 's1p1c1seed', 's1p1c1sval',
     's1p1c2qharv', 's1p1c2cons', 's1p1c2lives', 's1p1c2lost', 's1p1c2mkt', 's1p1c2sold', 's1p1c2seed', 's1p1c2sval',
     's1p1c3cons', 's1p1c3lost', 's1p1c3mkt', 's1p1c3sold', 's1p1c3seed', 's1p1c3sval',
     's1p1c4qharv', 's1p1c4cons', 's1p1c4lives', 's1p1c4lost', 's1p1c4mkt', 's1p1c4sold', 's1p1c4seed', 's1p1c4sval',
     's1p1c5cons', 's1p1c5lives', 's1p1c5lost', 's1p1c5mkt', 's1p1c5seed', 's1p1c5sval',
     's1p1c6qharv', 's1p1c6cons', 's1p1c6lives', 's1p1c6lost', 's1p1c6mkt', 's1p1c6sold', 's1p1c6seed', 's1p1c6sval']
]

# Group similar columns and aggregate them
aggregated_columns = selected_columns.groupby(selected_columns.columns, axis=1).sum()

# Select the first 10 aggregated columns
selected_columns_10 = aggregated_columns.iloc[:, :10]

# dump the columns into df
df[['spcq1', 'spcq2', 'spcq3', 'spcq4', 'spcq5', 'spcq6', 'spcq7', 'spcq8', 'spcq9', 'spcq10']] = selected_columns_10

C:\Users\DONKAMS\AppData\Local\Temp\ipykernel_15520\3218253916.py:12: FutureWarning: DataFrame.groupby with axis=1 is deprecated. Do `frame.T.groupby(...)` without axis instead.
  aggregated_columns = selected_columns.groupby(selected_columns.columns, axis=1).sum()


In [41]:
df

farmtype  TotalPlotArea LandSystem TenureType UsageYears    spcq1  \
1087       3.0         2.5000    3.0 0.0    1.0 0.0   16.0 0.0  24000.0   
1088       2.0         1.4417    2.0 2.0    1.0 1.0  33.0 33.0   1550.0   
1089       2.0         1.5250    3.0 3.0    1.0 1.0  33.0 33.0   2000.0   
1090       2.0         1.6626    3.0 3.0    1.0 1.0  33.0 33.0   1250.0   
1091       1.0         0.6250    2.0 2.0    1.0 1.0  25.0 25.0   1250.0   
...        ...            ...        ...        ...        ...      ...   
5474       2.0         5.0000    3.0 0.0    1.0 0.0    9.0 0.0    400.0   
5475       2.0         8.0000    1.0 0.0    1.0 0.0   11.0 0.0    600.0   
5476       2.0        17.0000    3.0 2.0    1.0 1.0  26.0 18.0   1200.0   
5477       2.0         8.0000    3.0 0.0    1.0 0.0    8.0 0.0    600.0   
5478       2.0        13.0000    3.0 2.0    1.0 1.0  15.0 12.0    600.0   

      spcq2  spcq3  spcq4   spcq5 spcq6   spcq7        spcq8  spcq9   spcq10  
1087    0.0    0.0    2.0  4800.0   140  2400.0            2    0.0  60000.0  
1088    0.0    0.0    2.0  2480.0    60   930.0          2.5    0.0  35000.0  
1089    0.0    0.0    2.0  2700.0    60   700.0  2.329999924  500.0      0.0  
1090    0.0    0.0    2.0  2250.0    60  1000.0          2.5    0.0  38000.0  
1091    0.0    0.0    2.0  2250.0    60  1000.0            2    0.0  14000.0  
...     ...    ...    ...     ...   ...     ...          ...    ...      ...  
5474    0.0    0.0    2.0  2000.0    10  1600.0          425  200.0      0.0  
5475    0.0    0.0    2.0  2800.0    10  2200.0          375  200.0      0.0  
5476    0.0    0.0    2.0  3200.0    20  2000.0          425  300.0      0.0  
5477    0.0    0.0    2.0  2000.0    10  1400.0          450  200.0      0.0  
5478    0.0    0.0    2.0  3000.0    16  2400.0          475  300.0      0.0  

[1800 rows x 15 columns]

In [42]:
#checking out columns from index 23 to 501
third_201_400_columns = filtered_data.iloc[:, 201:400]
third_201_400_columns_names = filtered_data.columns[201:400]

In [43]:
third_201_400_columns

s2p1c5harv  s2p1c5qharv  s2p1c5cons  s2p1c5lives  s2p1c5lost  s2p1c5mkt  \
1087        NaN          0.0         0.0          0.0         0.0        0.0   
1088        NaN          0.0         0.0          0.0         0.0        0.0   
1089        NaN          0.0         0.0          0.0         0.0        0.0   
1090        NaN          0.0         0.0          0.0         0.0        0.0   
1091        NaN          0.0         0.0          0.0         0.0        0.0   
...         ...          ...         ...          ...         ...        ...   
5474        NaN          0.0         0.0          0.0         0.0        0.0   
5475        NaN          0.0         0.0          0.0         0.0        0.0   
5476        NaN          0.0         0.0          0.0         0.0        0.0   
5477        NaN          0.0         0.0          0.0         0.0        0.0   
5478        NaN          0.0         0.0          0.0         0.0        0.0   

      s2p1c5sold  s2p1c5seed  s2p1c5sval s2p1c6plant s2p1c6harv  s2p1c6qharv  \
1087         0.0         NaN         0.0         NaN        NaN          0.0   
1088         0.0         NaN         0.0         NaN        NaN          0.0   
1089         0.0         NaN         0.0         NaN        NaN          0.0   
1090         0.0         NaN         0.0         NaN        NaN          0.0   
1091         0.0         NaN         0.0         NaN        NaN          0.0   
...          ...         ...         ...         ...        ...          ...   
5474         0.0         0.0         0.0         NaN        NaN          0.0   
5475         0.0         0.0         0.0         NaN        NaN          0.0   
5476         0.0         0.0         0.0         NaN        NaN          0.0   
5477         0.0         0.0         0.0         NaN        NaN          0.0   
5478         0.0         0.0         0.0         NaN        NaN          0.0   

      s2p1c6cons  s2p1c6lives  s2p1c6lost  s2p1c6mkt  s2p1c6sold  s2p1c6seed  \
1087         0.0          0.0         0.0        0.0         0.0         NaN   
1088         0.0          0.0         0.0        0.0         0.0         NaN   
1089         0.0          0.0         0.0        0.0         0.0         NaN   
1090         0.0          0.0         0.0        0.0         0.0         NaN   
1091         0.0          0.0         0.0        0.0         0.0         NaN   
...          ...          ...         ...        ...         ...         ...   
5474         0.0          0.0         0.0        0.0         0.0         0.0   
5475         0.0          0.0         0.0        0.0         0.0         0.0   
5476         0.0          0.0         0.0        0.0         0.0         0.0   
5477         0.0          0.0         0.0        0.0         0.0         0.0   
5478         0.0          0.0         0.0        0.0         0.0         0.0   

      s2p1c6sval s2p2c1plant s2p2c1harv  s2p2c1qharv  s2p2c1cons  s2p2c1lives  \
1087         0.0         NaN        NaN          0.0         0.0          0.0   
1088         0.0         NaN        NaN          0.0         0.0          0.0   
1089         0.0         NaN        NaN          0.0         0.0          0.0   
1090         0.0         NaN        NaN          0.0         0.0          0.0   
1091         0.0         NaN        NaN          0.0         0.0          0.0   
...          ...         ...        ...          ...         ...          ...   
5474         0.0           0          0          0.0         0.0          0.0   
5475         0.0           0          0          0.0         0.0          0.0   
5476         0.0           0          0          0.0         0.0          0.0   
5477         0.0           0          0          0.0         0.0          0.0   
5478         0.0           0          0          0.0         0.0          0.0   

      s2p2c1lost  s2p2c1mkt  s2p2c1sold s2p2c1seed s2p2c1sval s2p2c2plant  \
1087         0.0        0.0         0.0          0          0         Na

In [44]:
# Remove columns with missing values
third_201_400_columns_cleaned = third_201_400_columns.dropna(axis=1)

# Now, third_201_400_columns_cleaned contains the DataFrame with missing columns removed


In [45]:
third_201_400_columns_cleaned

s2p1c5qharv  s2p1c5cons  s2p1c5lives  s2p1c5lost  s2p1c5mkt  s2p1c5sold  \
1087          0.0         0.0          0.0         0.0        0.0         0.0   
1088          0.0         0.0          0.0         0.0        0.0         0.0   
1089          0.0         0.0          0.0         0.0        0.0         0.0   
1090          0.0         0.0          0.0         0.0        0.0         0.0   
1091          0.0         0.0          0.0         0.0        0.0         0.0   
...           ...         ...          ...         ...        ...         ...   
5474          0.0         0.0          0.0         0.0        0.0         0.0   
5475          0.0         0.0          0.0         0.0        0.0         0.0   
5476          0.0         0.0          0.0         0.0        0.0         0.0   
5477          0.0         0.0          0.0         0.0        0.0         0.0   
5478          0.0         0.0          0.0         0.0        0.0         0.0   

      s2p1c5sval  s2p1c6qharv  s2p1c6cons  s2p1c6lives  s2p1c6lost  s2p1c6mkt  \
1087         0.0          0.0         0.0          0.0         0.0        0.0   
1088         0.0          0.0         0.0          0.0         0.0        0.0   
1089         0.0          0.0         0.0          0.0         0.0        0.0   
1090         0.0          0.0         0.0          0.0         0.0        0.0   
1091         0.0          0.0         0.0          0.0         0.0        0.0   
...          ...          ...         ...          ...         ...        ...   
5474         0.0          0.0         0.0          0.0         0.0        0.0   
5475         0.0          0.0         0.0          0.0         0.0        0.0   
5476         0.0          0.0         0.0          0.0         0.0        0.0   
5477         0.0          0.0         0.0          0.0         0.0        0.0   
5478         0.0          0.0         0.0          0.0         0.0        0.0   

      s2p1c6sold  s2p1c6sval  s2p2c1qharv  s2p2c1cons  s2p2c1lives  \
1087         0.0         0.0          0.0         0.0          0.0   
1088         0.0         0.0          0.0         0.0          0.0   
1089         0.0         0.0          0.0         0.0          0.0   
1090         0.0         0.0          0.0         0.0          0.0   
1091         0.0         0.0          0.0         0.0          0.0   
...          ...         ...          ...         ...          ...   
5474         0.0         0.0          0.0         0.0          0.0   
5475         0.0         0.0          0.0         0.0          0.0   
5476         0.0         0.0          0.0         0.0          0.0   
5477         0.0         0.0          0.0         0.0          0.0   
5478         0.0         0.0          0.0         0.0          0.0   

      s2p2c1lost  s2p2c1mkt  s2p2c1sold s2p2c1seed s2p2c1sval  s2p2c2qharv  \
1087         0.0        0.0         0.0          0          0          0.0   
1088         0.0        0.0         0.0          0          0          0.0   
1089         0.0        0.0         0.0          0          0          0.0   
1090         0.0        0.0         0.0          0          0          0.0   
1091         0.0        0.0         0.0          0          0          0.0   
...          ...        ...         ...        ...        ...          ...   
5474         0.0        0.0         0.0          0          0          0.0   
5475         0.0        0.0         0.0          0          0          0.0   
5476         0.0        0.0         0.0          0          0          0.0   
5477         0.0        0.0         0.0          0          0          0.0   
5478         0.0        0.0         0.0          0          0          0.0   

      s2p2c2cons  s2p2c2lives  s2p2c2lost  s2p2c2mkt  s2p2c2sold s2p2c2seed  \
1087         0.0          0.0         0.0        0.0         0.0          0   
1088         0.0          0.0         0.0        0.0         0.0          0   
1089         0.0          0.0         0.0    

In [51]:

# Select only numeric columns
numeric_columns = third_201_400_columns.select_dtypes(include=[float, int])

# Group and aggregate columns by similar names (you can adjust this as needed)
aggregated_data = numeric_columns.groupby(numeric_columns.columns.str.extract('([a-z]+)', expand=False), axis=1).mean()

# Select only the first 10 columns
result = aggregated_data.iloc[:, :10]

# result now contains the DataFrame with reduced columns
result

C:\Users\DONKAMS\AppData\Local\Temp\ipykernel_15520\257799533.py:5: FutureWarning: DataFrame.groupby with axis=1 is deprecated. Do `frame.T.groupby(...)` without axis instead.
  aggregated_data = numeric_columns.groupby(numeric_columns.columns.str.extract('([a-z]+)', expand=False), axis=1).mean()


nyieldc    pc           s
1087   6750.0  33.4   42.549618
1088   9940.0  48.6  483.366412
1089   6200.0  46.8   81.458015
1090  10380.0  48.6   34.895038
1091   4300.0  29.8   14.206107
...       ...   ...         ...
5474    220.0   7.8    0.785714
5475    260.0  11.0    2.242857
5476    440.0  22.4    3.678571
5477    180.0   4.8    1.521429
5478    340.0  13.6    2.950000

[1800 rows x 3 columns]

In [52]:
df[['nyieldc', 'pc', 's']] = result

In [53]:
df

farmtype  TotalPlotArea LandSystem TenureType UsageYears    spcq1  \
1087       3.0         2.5000    3.0 0.0    1.0 0.0   16.0 0.0  24000.0   
1088       2.0         1.4417    2.0 2.0    1.0 1.0  33.0 33.0   1550.0   
1089       2.0         1.5250    3.0 3.0    1.0 1.0  33.0 33.0   2000.0   
1090       2.0         1.6626    3.0 3.0    1.0 1.0  33.0 33.0   1250.0   
1091       1.0         0.6250    2.0 2.0    1.0 1.0  25.0 25.0   1250.0   
...        ...            ...        ...        ...        ...      ...   
5474       2.0         5.0000    3.0 0.0    1.0 0.0    9.0 0.0    400.0   
5475       2.0         8.0000    1.0 0.0    1.0 0.0   11.0 0.0    600.0   
5476       2.0        17.0000    3.0 2.0    1.0 1.0  26.0 18.0   1200.0   
5477       2.0         8.0000    3.0 0.0    1.0 0.0    8.0 0.0    600.0   
5478       2.0        13.0000    3.0 2.0    1.0 1.0  15.0 12.0    600.0   

      spcq2  spcq3  spcq4   spcq5 spcq6   spcq7        spcq8  spcq9   spcq10  \
1087    0.0    0.0    2.0  4800.0   140  2400.0            2    0.0  60000.0   
1088    0.0    0.0    2.0  2480.0    60   930.0          2.5    0.0  35000.0   
1089    0.0    0.0    2.0  2700.0    60   700.0  2.329999924  500.0      0.0   
1090    0.0    0.0    2.0  2250.0    60  1000.0          2.5    0.0  38000.0   
1091    0.0    0.0    2.0  2250.0    60  1000.0            2    0.0  14000.0   
...     ...    ...    ...     ...   ...     ...          ...    ...      ...   
5474    0.0    0.0    2.0  2000.0    10  1600.0          425  200.0      0.0   
5475    0.0    0.0    2.0  2800.0    10  2200.0          375  200.0      0.0   
5476    0.0    0.0    2.0  3200.0    20  2000.0          425  300.0      0.0   
5477    0.0    0.0    2.0  2000.0    10  1400.0          450  200.0      0.0   
5478    0.0    0.0    2.0  3000.0    16  2400.0          475  300.0      0.0   

      nyieldc    pc           s  
1087   6750.0  33.4   42.549618  
1088   9940.0  48.6  483.366412  
1089   6200.0  46.8   81.458015  
1090  10380.0  48.6   34.895038  
1091   4300.0  29.8   14.206107  
...       ...   ...         ...  
5474    220.0   7.8    0.785714  
5475    260.0  11.0    2.242857  
5476    440.0  22.4    3.678571  
5477    180.0   4.8    1.521429  
5478    340.0  13.6    2.950000  

[1800 rows x 18 columns]

In [61]:
#checking out columns from index 401 to 600
fourth_401_600_columns = filtered_data.iloc[:, 401:600]
fourth_401_600_columns_names = filtered_data.columns[401:600]

In [62]:
fourth_401_600_columns

s3p2irrig4  s3p2pest  s3p2wat1  s3p2wat2  s3p2wat3  s3p2wat4  s3p2wat5  \
1087         0.0       0.0       0.0       0.0       0.0       0.0       0.0   
1088         0.0       0.0       0.0       0.0       0.0       0.0       0.0   
1089         0.0       0.0       0.0       0.0       0.0       0.0       0.0   
1090         0.0       0.0       0.0       0.0       0.0       0.0       0.0   
1091         0.0       0.0       0.0       0.0       0.0       0.0       0.0   
...          ...       ...       ...       ...       ...       ...       ...   
5474         0.0       0.0       0.0       0.0       0.0       0.0       0.0   
5475         0.0       0.0       0.0       0.0       0.0       0.0       0.0   
5476         0.0       0.0       0.0       0.0       0.0       0.0       0.0   
5477         0.0       0.0       0.0       0.0       0.0       0.0       0.0   
5478         0.0       0.0       0.0       0.0       0.0       0.0       0.0   

      lvsown  lvs1born  lvs1graze1  lvs1graze2  lvs1graze3  lvs1lost  lvs1num  \
1087     1.0       0.0         0.0         0.0         0.0       0.0      0.0   
1088     1.0       0.0         0.0         0.0         0.0       0.0      0.0   
1089     1.0       2.0         0.0         0.0         0.0       0.0      4.0   
1090     1.0       0.0         0.0         0.0         0.0       0.0      0.0   
1091     1.0       0.0         0.0         0.0         0.0       0.0      0.0   
...      ...       ...         ...         ...         ...       ...      ...   
5474     1.0       0.0         0.0         0.0         0.0       0.0      0.0   
5475     1.0       0.0         0.0         0.0         0.0       0.0      0.0   
5476     1.0       0.0         0.0         0.0         0.0       0.0      0.0   
5477     1.0       0.0         0.0         0.0         0.0       0.0      0.0   
5478     1.0       0.0         0.0         0.0         0.0       0.0      0.0   

      lvs1pprice  lvs1purch  lvs1sold  lvs1sprice  lvs2born  lvs2graze1  \
1087         0.0        0.0       0.0         0.0       2.0         0.0   
1088         0.0        0.0       0.0         0.0       1.0         0.0   
1089         0.0        0.0       1.0      2000.0       0.0         0.0   
1090         0.0        0.0       0.0         0.0       1.0         0.0   
1091         0.0        0.0       0.0         0.0       1.0         0.0   
...          ...        ...       ...         ...       ...         ...   
5474         0.0        0.0       0.0         0.0       0.0         0.0   
5475         0.0        0.0       0.0         0.0       0.0         0.0   
5476         0.0        0.0       7.0      3000.0       0.0         6.0   
5477         0.0        0.0       3.0     51000.0       0.0         0.0   
5478         0.0        0.0       0.0         0.0       0.0         0.0   

      lvs2graze2  lvs2graze3  lvs2lost  lvs2num  lvs2pprice  lvs2purch  \
1087         0.0         0.0       0.0      5.0      3000.0        2.0   
1088         0.0         0.0       0.0      2.0      3800.0        1.0   
1089         0.0         0.0       0.0      0.0         0.0        0.0   
1090         0.0         0.0       0.0      2.0      2700.0        1.0   
1091         6.0         0.0       0.0      2.0      3600.0        1.0   
...          ...         ...       ...      ...         ...        ...   
5474         3.0         9.0       0.0      0.0         0.0        0.0   
5475         0.0        12.0       3.0      7.0         0.0        9.0   
5476         0.0         8.0       0.0      0.0         6.0        0.0   
5477         0.0         0.0       0.0      0.0         6.0        0.0   
5478         3.0         9.0       1.0      1.0         0.0        0.0   

      lvs2sold  lvs2sprice  lvs3born  lvs3graze1  lvs3graze2  lvs3graze3  \
1087       7.0      3500.0       0.0         0.0         0.0         0.0   
1088       0.0         0.0       0.0         0.0         0.0         0.0   
1089       0.0         0.0       0.0         0.0      

In [63]:
fourth_401_600_columns_names

Index(['s3p2irrig4', 's3p2pest', 's3p2wat1', 's3p2wat2', 's3p2wat3',
       's3p2wat4', 's3p2wat5', 'lvsown', 'lvs1born', 'lvs1graze1',
       ...
       'ad755', 'ad756', 'ad757', 'ad758', 'ad759', 'ad7510', 'ad7511',
       'ad7512', 'ad7513', 'ad7514'],
      dtype='object', length=199)

In [64]:
# Remove columns with missing values
fourth_401_600_columns = fourth_401_600_columns.dropna(axis=1)

In [65]:
fourth_401_600_columns

s3p2pest  s3p2wat1  s3p2wat2  s3p2wat3  s3p2wat4  s3p2wat5  lvsown  \
1087       0.0       0.0       0.0       0.0       0.0       0.0     1.0   
1088       0.0       0.0       0.0       0.0       0.0       0.0     1.0   
1089       0.0       0.0       0.0       0.0       0.0       0.0     1.0   
1090       0.0       0.0       0.0       0.0       0.0       0.0     1.0   
1091       0.0       0.0       0.0       0.0       0.0       0.0     1.0   
...        ...       ...       ...       ...       ...       ...     ...   
5474       0.0       0.0       0.0       0.0       0.0       0.0     1.0   
5475       0.0       0.0       0.0       0.0       0.0       0.0     1.0   
5476       0.0       0.0       0.0       0.0       0.0       0.0     1.0   
5477       0.0       0.0       0.0       0.0       0.0       0.0     1.0   
5478       0.0       0.0       0.0       0.0       0.0       0.0     1.0   

      lvs1born  lvs1graze1  lvs1graze2  lvs1graze3  lvs1lost  lvs1num  \
1087       0.0         0.0         0.0         0.0       0.0      0.0   
1088       0.0         0.0         0.0         0.0       0.0      0.0   
1089       2.0         0.0         0.0         0.0       0.0      4.0   
1090       0.0         0.0         0.0         0.0       0.0      0.0   
1091       0.0         0.0         0.0         0.0       0.0      0.0   
...        ...         ...         ...         ...       ...      ...   
5474       0.0         0.0         0.0         0.0       0.0      0.0   
5475       0.0         0.0         0.0         0.0       0.0      0.0   
5476       0.0         0.0         0.0         0.0       0.0      0.0   
5477       0.0         0.0         0.0         0.0       0.0      0.0   
5478       0.0         0.0         0.0         0.0       0.0      0.0   

      lvs1pprice  lvs1purch  lvs1sold  lvs2born  lvs2graze3  lvs2lost  \
1087         0.0        0.0       0.0       2.0         0.0       0.0   
1088         0.0        0.0       0.0       1.0         0.0       0.0   
1089         0.0        0.0       1.0       0.0         0.0       0.0   
1090         0.0        0.0       0.0       1.0         0.0       0.0   
1091         0.0        0.0       0.0       1.0         0.0       0.0   
...          ...        ...       ...       ...         ...       ...   
5474         0.0        0.0       0.0       0.0         9.0       0.0   
5475         0.0        0.0       0.0       0.0        12.0       3.0   
5476         0.0        0.0       7.0       0.0         8.0       0.0   
5477         0.0        0.0       3.0       0.0         0.0       0.0   
5478         0.0        0.0       0.0       0.0         9.0       1.0   

      lvs2num  lvs2pprice  lvs2purch  lvs2sold  lvs2sprice  lvs3born  \
1087      5.0      3000.0        2.0       7.0      3500.0       0.0   
1088      2.0      3800.0        1.0       0.0         0.0       0.0   
1089      0.0         0.0        0.0       0.0         0.0       0.0   
1090      2.0      2700.0        1.0       0.0         0.0       0.0   
1091      2.0      3600.0        1.0       0.0         0.0       0.0   
...       ...         ...        ...       ...         ...       ...   
5474      0.0         0.0        0.0       0.0         0.0       0.0   
5475      7.0         0.0        9.0       0.0         0.0       0.0   
5476      0.0         6.0        0.0       0.0         0.0       0.0   
5477      0.0         6.0        0.0       3.0         0.0       0.0   
5478      1.0         0.0        0.0       0.0         0.0       0.0   

      lvs3graze1  lvs3graze2  lvs3lost  lvs3num  lvs3pprice  lvs3purch  \
1087         0.0         0.0       0.0      2.0         0.0        0.0   
1088         0.0         0.0       0.0      0.0         0.0        0.0   
1089         0.0         0.0       0.0      0.0         0.0        0.0   
1090         0.0         0.0       0.0      0.0         0.0        0.0   
1091         0.0         0.0       0.0      0.0         0.0        0.0   
...          ...         ...  

In [66]:
# Select only numeric columns
numeric_columns = fourth_401_600_columns.select_dtypes(include=[float, int])

# Group and aggregate columns by similar names (you can adjust this as needed)
aggregated_data = numeric_columns.groupby(numeric_columns.columns.str.extract('([a-z]+)', expand=False), axis=1).mean()

# Select only the first 10 columns
result = aggregated_data.iloc[:, :10]

# result now contains the DataFrame with reduced columns
result

C:\Users\DONKAMS\AppData\Local\Temp\ipykernel_15520\3510058525.py:5: FutureWarning: DataFrame.groupby with axis=1 is deprecated. Do `frame.T.groupby(...)` without axis instead.
  aggregated_data = numeric_columns.groupby(numeric_columns.columns.str.extract('([a-z]+)', expand=False), axis=1).mean()


ad     cost          lvs  lvsown   lvsp    s
1087  0.070175    62.50    88.500000     1.0  38.03  0.0
1088  0.122807    62.50    51.405405     1.0  40.00  0.0
1089  0.087719    43.75     0.094595     1.0  48.00  0.0
1090  0.087719    50.00    36.540541     1.0  80.00  0.0
1091  0.087719    37.50    48.891892     1.0  32.00  0.0
...        ...      ...          ...     ...    ...  ...
5474  0.070175   950.00   108.716216     1.0   4.48  0.0
5475  0.070175  1250.00  1285.202703     1.0  22.80  0.0
5476  0.087719  2875.00   102.378378     1.0  22.40  0.0
5477  0.070175   850.00     0.337838     1.0   8.00  0.0
5478  0.087719     0.00    95.027027     1.0   8.40  0.0

[1800 rows x 6 columns]

In [67]:
df[['ad','cost','lvs', 'lvsown',	'lvsp',	's2']] = result

In [68]:
df

farmtype  TotalPlotArea LandSystem TenureType UsageYears    spcq1  \
1087       3.0         2.5000    3.0 0.0    1.0 0.0   16.0 0.0  24000.0   
1088       2.0         1.4417    2.0 2.0    1.0 1.0  33.0 33.0   1550.0   
1089       2.0         1.5250    3.0 3.0    1.0 1.0  33.0 33.0   2000.0   
1090       2.0         1.6626    3.0 3.0    1.0 1.0  33.0 33.0   1250.0   
1091       1.0         0.6250    2.0 2.0    1.0 1.0  25.0 25.0   1250.0   
...        ...            ...        ...        ...        ...      ...   
5474       2.0         5.0000    3.0 0.0    1.0 0.0    9.0 0.0    400.0   
5475       2.0         8.0000    1.0 0.0    1.0 0.0   11.0 0.0    600.0   
5476       2.0        17.0000    3.0 2.0    1.0 1.0  26.0 18.0   1200.0   
5477       2.0         8.0000    3.0 0.0    1.0 0.0    8.0 0.0    600.0   
5478       2.0        13.0000    3.0 2.0    1.0 1.0  15.0 12.0    600.0   

      spcq2  spcq3  spcq4   spcq5 spcq6   spcq7        spcq8  spcq9   spcq10  \
1087    0.0    0.0    2.0  4800.0   140  2400.0            2    0.0  60000.0   
1088    0.0    0.0    2.0  2480.0    60   930.0          2.5    0.0  35000.0   
1089    0.0    0.0    2.0  2700.0    60   700.0  2.329999924  500.0      0.0   
1090    0.0    0.0    2.0  2250.0    60  1000.0          2.5    0.0  38000.0   
1091    0.0    0.0    2.0  2250.0    60  1000.0            2    0.0  14000.0   
...     ...    ...    ...     ...   ...     ...          ...    ...      ...   
5474    0.0    0.0    2.0  2000.0    10  1600.0          425  200.0      0.0   
5475    0.0    0.0    2.0  2800.0    10  2200.0          375  200.0      0.0   
5476    0.0    0.0    2.0  3200.0    20  2000.0          425  300.0      0.0   
5477    0.0    0.0    2.0  2000.0    10  1400.0          450  200.0      0.0   
5478    0.0    0.0    2.0  3000.0    16  2400.0          475  300.0      0.0   

      nyieldc    pc           s        ad     cost          lvs  lvsown  \
1087   6750.0  33.4   42.549618  0.070175    62.50    88.500000     1.0   
1088   9940.0  48.6  483.366412  0.122807    62.50    51.405405     1.0   
1089   6200.0  46.8   81.458015  0.087719    43.75     0.094595     1.0   
1090  10380.0  48.6   34.895038  0.087719    50.00    36.540541     1.0   
1091   4300.0  29.8   14.206107  0.087719    37.50    48.891892     1.0   
...       ...   ...         ...       ...      ...          ...     ...   
5474    220.0   7.8    0.785714  0.070175   950.00   108.716216     1.0   
5475    260.0  11.0    2.242857  0.070175  1250.00  1285.202703     1.0   
5476    440.0  22.4    3.678571  0.087719  2875.00   102.378378     1.0   
5477    180.0   4.8    1.521429  0.070175   850.00     0.337838     1.0   
5478    340.0  13.6    2.950000  0.087719     0.00    95.027027     1.0   

       lvsp   s2  
1087  38.03  0.0  
1088  40.00  0.0  
1089  48.00  0.0  
1090  80.00  0.0  
1091  32.00  0.0  
...     ...  ...  
5474   4.48  0.0  
5475  22.80  0.0  
5476  22.40  0.0  
5477   8.00  0.0  
5478   8.40  0.0  

[1800 rows x 24 columns]

In [70]:
#checking out columns from index 401 to 600
last_601_823_columns = filtered_data.iloc[:, 601:823]
last_601_823_columns_names = filtered_data.columns[601:823]

In [71]:
last_601_823_columns

ad7516  ad7517  ad7518  ad7519  ad7520 ad7521  ad7522  ad7523  ad7524  \
1087     0.0     0.0     1.0     0.0     0.0      0     0.0     0.0     0.0   
1088     0.0     0.0     0.0     0.0     0.0      0     0.0     0.0     0.0   
1089     0.0     0.0     0.0     0.0     0.0      0     0.0     0.0     0.0   
1090     0.0     0.0     0.0     0.0     0.0      0     0.0     0.0     0.0   
1091     0.0     0.0     0.0     0.0     0.0      0     0.0     0.0     0.0   
...      ...     ...     ...     ...     ...    ...     ...     ...     ...   
5474     0.0     0.0     0.0     0.0     0.0      0     0.0     0.0     0.0   
5475     0.0     0.0     0.0     0.0     0.0      0     0.0     0.0     0.0   
5476     0.0     0.0     0.0     0.0     0.0      0     0.0     0.0     0.0   
5477     0.0     0.0     0.0     0.0     0.0      0     0.0     0.0     0.0   
5478     0.0     0.0     0.0     0.0     0.0      0     0.0     0.0     0.0   

      ad7525  ad761  ad762  ad763  ad764  ad765  ad766  ad767  ad768  ad769  \
1087     0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0   
1088     0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0   
1089     0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0   
1090     0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0   
1091     0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0   
...      ...    ...    ...    ...    ...    ...    ...    ...    ...    ...   
5474     1.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0   
5475     1.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0   
5476     1.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0   
5477     1.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0   
5478     1.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0    0.0   

      ad7610  ad7611  ad7612  ad7613  ad7614  ad7615  ad7616  ad7617  ad7618  \
1087     0.0     0.0     1.0     0.0     0.0     0.0     0.0     0.0     0.0   
1088     0.0     0.0     0.0     0.0     1.0     1.0     0.0     1.0     0.0   
1089     0.0     0.0     0.0     0.0     1.0     1.0     0.0     1.0     0.0   
1090     0.0     0.0     0.0     0.0     1.0     1.0     0.0     1.0     0.0   
1091     0.0     0.0     0.0     0.0     0.0     0.0     0.0     0.0     0.0   
...      ...     ...     ...     ...     ...     ...     ...     ...     ...   
5474     0.0     0.0     0.0     0.0     0.0     0.0     0.0     0.0     1.0   
5475     0.0     0.0     0.0     0.0     0.0     0.0     0.0     0.0     1.0   
5476     0.0     0.0     0.0     0.0     0.0     0.0     0.0     0.0     1.0   
5477     0.0     0.0     0.0     0.0     0.0     0.0     0.0     0.0     1.0   
5478     0.0     0.0     0.0     0.0     0.0     0.0     0.0     0.0     1.0   

      ad7619  ad7620  ad7621  ad7622  ad7623 ad7624  ad7625  
1087     0.0     0.0     0.0     0.0     0.0      0     0.0  
1088     0.0     0.0     0.0     0.0     0.0      0     0.0  
1089     0.0     0.0     0.0     0.0     0.0      0     0.0  
1090     0.0     0.0     0.0     0.0     0.0      0     0.0  
1091     0.0     0.0     0.0     1.0     0.0      0     0.0  
...      ...     ...     ...     ...     ...    ...     ...  
5474     0.0     0.0     0.0     0.0     0.0      0     0.0  
5475     0.0     0.0     0.0     0.0     0.0      0     0.0  
5476     0.0     0.0     0.0     0.0     0.0      0     0.0  
5477     0.0     0.0     0.0     0.0     0.0      0     0.0  
5478     0.0     0.0     0.0     0.0     0.0      0     0.0  

[1800 rows x 35 columns]

In [72]:
# Select only numeric columns
numeric_columns = last_601_823_columns.select_dtypes(include=[float, int])

# Group and aggregate columns by similar names (you can adjust this as needed)
aggregated_data = numeric_columns.groupby(numeric_columns.columns.str.extract('([a-z]+)', expand=False), axis=1).mean()

# Select only the first 10 columns
result = aggregated_data.iloc[:, :10]

# result now contains the DataFrame with reduced columns
result

C:\Users\DONKAMS\AppData\Local\Temp\ipykernel_15520\3487283524.py:5: FutureWarning: DataFrame.groupby with axis=1 is deprecated. Do `frame.T.groupby(...)` without axis instead.
  aggregated_data = numeric_columns.groupby(numeric_columns.columns.str.extract('([a-z]+)', expand=False), axis=1).mean()


ad
1087  0.060606
1088  0.090909
1089  0.090909
1090  0.090909
1091  0.030303
...        ...
5474  0.060606
5475  0.060606
5476  0.060606
5477  0.060606
5478  0.060606

[1800 rows x 1 columns]

In [73]:
df['ads'] = result

In [74]:
df

farmtype  TotalPlotArea LandSystem TenureType UsageYears    spcq1  \
1087       3.0         2.5000    3.0 0.0    1.0 0.0   16.0 0.0  24000.0   
1088       2.0         1.4417    2.0 2.0    1.0 1.0  33.0 33.0   1550.0   
1089       2.0         1.5250    3.0 3.0    1.0 1.0  33.0 33.0   2000.0   
1090       2.0         1.6626    3.0 3.0    1.0 1.0  33.0 33.0   1250.0   
1091       1.0         0.6250    2.0 2.0    1.0 1.0  25.0 25.0   1250.0   
...        ...            ...        ...        ...        ...      ...   
5474       2.0         5.0000    3.0 0.0    1.0 0.0    9.0 0.0    400.0   
5475       2.0         8.0000    1.0 0.0    1.0 0.0   11.0 0.0    600.0   
5476       2.0        17.0000    3.0 2.0    1.0 1.0  26.0 18.0   1200.0   
5477       2.0         8.0000    3.0 0.0    1.0 0.0    8.0 0.0    600.0   
5478       2.0        13.0000    3.0 2.0    1.0 1.0  15.0 12.0    600.0   

      spcq2  spcq3  spcq4   spcq5 spcq6   spcq7        spcq8  spcq9   spcq10  \
1087    0.0    0.0    2.0  4800.0   140  2400.0            2    0.0  60000.0   
1088    0.0    0.0    2.0  2480.0    60   930.0          2.5    0.0  35000.0   
1089    0.0    0.0    2.0  2700.0    60   700.0  2.329999924  500.0      0.0   
1090    0.0    0.0    2.0  2250.0    60  1000.0          2.5    0.0  38000.0   
1091    0.0    0.0    2.0  2250.0    60  1000.0            2    0.0  14000.0   
...     ...    ...    ...     ...   ...     ...          ...    ...      ...   
5474    0.0    0.0    2.0  2000.0    10  1600.0          425  200.0      0.0   
5475    0.0    0.0    2.0  2800.0    10  2200.0          375  200.0      0.0   
5476    0.0    0.0    2.0  3200.0    20  2000.0          425  300.0      0.0   
5477    0.0    0.0    2.0  2000.0    10  1400.0          450  200.0      0.0   
5478    0.0    0.0    2.0  3000.0    16  2400.0          475  300.0      0.0   

      nyieldc    pc           s        ad     cost          lvs  lvsown  \
1087   6750.0  33.4   42.549618  0.070175    62.50    88.500000     1.0   
1088   9940.0  48.6  483.366412  0.122807    62.50    51.405405     1.0   
1089   6200.0  46.8   81.458015  0.087719    43.75     0.094595     1.0   
1090  10380.0  48.6   34.895038  0.087719    50.00    36.540541     1.0   
1091   4300.0  29.8   14.206107  0.087719    37.50    48.891892     1.0   
...       ...   ...         ...       ...      ...          ...     ...   
5474    220.0   7.8    0.785714  0.070175   950.00   108.716216     1.0   
5475    260.0  11.0    2.242857  0.070175  1250.00  1285.202703     1.0   
5476    440.0  22.4    3.678571  0.087719  2875.00   102.378378     1.0   
5477    180.0   4.8    1.521429  0.070175   850.00     0.337838     1.0   
5478    340.0  13.6    2.950000  0.087719     0.00    95.027027     1.0   

       lvsp   s2       ads  
1087  38.03  0.0  0.060606  
1088  40.00  0.0  0.090909  
1089  48.00  0.0  0.090909  
1090  80.00  0.0  0.090909  
1091  32.00  0.0  0.030303  
...     ...  ...       ...  
5474   4.48  0.0  0.060606  
5475  22.80  0.0  0.060606  
5476  22.40  0.0  0.060606  
5477   8.00  0.0  0.060606  
5478   8.40  0.0  0.060606  

[1800 rows x 25 columns]

In [75]:
df.shape

(1800, 25)